pkg/eventing/manager_test.go

```diff
@@ -0,0 +1,531 @@
+				*natstestutils.NewNATSCR(
+					natstestutils.WithNATSCRName("test-nats"),
+					natstestutils.WithNATSCRNamespace("test-namespace"),
+				),
+			},
+			givenNamespace: "test-namespace",
+			want:           "nats://test-nats.test-namespace.svc.cluster.local:4222",
+			wantErr:        nil,
+		},
+		{
+			name:                "NATS resource doesn't exist",
+			givenNatsResources:  []natsv1alpha1.NATS{},
+			givenNamespace:      "test-namespace",
+			want:                "",
+			getNATSResourcesErr: nil,
+			wantErr:             fmt.Errorf("NATS CR is not found to build NATS server URL"),
+		},
+		{
+			name:                "NATS resource does not exist",
+			givenNatsResources:  nil,
+			givenNamespace:      "test-namespace",
+			want:                "",
+			getNATSResourcesErr: fmt.Errorf("NATS CR is not found to build NATS server URL"),
+			wantErr:             fmt.Errorf("NATS CR is not found to build NATS server URL"),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			ctx := context.Background()
+			kubeClient := new(k8smocks.Client)
+			kubeClient.On("GetNATSResources", ctx, tc.givenNamespace).Return(&natsv1alpha1.NATSList{
+				Items: tc.givenNatsResources,
+			}, tc.getNATSResourcesErr)
+
+			em := EventingManager{
+				kubeClient: kubeClient,
+			}
+
+			// when
+			url, err := em.getNATSUrl(ctx, tc.givenNamespace)
+
+			// then
+			require.Equal(t, tc.wantErr, err)
+			require.Equal(t, tc.want, url)
+		})
+	}
+}
+
+func Test_UpdateNatsConfig(t *testing.T) {
+	// Define a list of test cases
+	testCases := []struct {
+		name               string
+		eventing           *v1alpha1.Eventing
+		expectedConfig     env.NATSConfig
+		givenNatsResources []natsv1alpha1.NATS
+		expectedError      error
+	}{
+		{
+			name: "Update NATSConfig",
+			eventing: testutils.NewEventingCR(
+				testutils.WithEventingCRName("test-eventing"),
+				testutils.WithEventingCRNamespace("test-namespace"),
+				testutils.WithEventingCRDefaults(),
+				testutils.WithEventingStreamData("File", "1Gi", "700Mi", 2, 1000),
+			),
+			givenNatsResources: []natsv1alpha1.NATS{
+				*natstestutils.NewNATSCR(
+					natstestutils.WithNATSCRName("test-nats"),
+					natstestutils.WithNATSCRNamespace("test-namespace"),
+				),
+			},
+			expectedConfig: env.NATSConfig{
+				URL:                     "nats://test-nats.test-namespace.svc.cluster.local:4222",
+				JSStreamStorageType:     "File",
+				JSStreamReplicas:        2,
+				JSStreamMaxBytes:        "700Mi",
+				JSStreamMaxMsgsPerTopic: 1000,
+			},
+			expectedError: nil,
+		},
+		{
+			name: "Error getting NATS URL",
+			eventing: testutils.NewEventingCR(
+				testutils.WithEventingCRName("test-eventing"),
+				testutils.WithEventingCRNamespace("test-namespace"),
+				testutils.WithEventingCRDefaults(),
+				testutils.WithEventingStreamData("Memory", "1Gi", "700Mi", 2, 1000),
+			),
+			expectedError: fmt.Errorf("failed to get NATS URL"),
+		},
```

Comment:
(nit) Please set the `givenNatsResources` to nil or empty value to make the test case more explicit and readable.